weather.py

```diff
@@ -0,0 +1,440 @@
+                                                         low_date, humid_date)
+            if not exists:
+                raise Exception("Files for this year are not in the system.")
+
+        else:
+            mon = int(date[1])  # Getting month in word form
+            if 0 < mon < 13:  # Validate month is in between 1-12
+                month = self.months[mon - 1]
+
+                _, high_temp, high_date, low_temp, low_date, high_humidity, \
+                humid_date = self.highest_per_month(date[0], month, high_temp,
+                                                    low_temp, high_humidity, high_date,
+                                                    low_date, humid_date)
+
+        results = dict()
+        # Storing the results obtained into a data structure defined in class level dictionaries
+        results["maxtemp"] = high_temp
+        results["mintemp"] = low_temp
+        results["maxdate"] = high_date
+        results["mindate"] = low_date
+        results["humid"] = high_humidity
+        results["humiddate"] = humid_date
+
+        return results
+
+    def calculate_average_helper(self, dic, avg_max_temp, count_max_temp,
+                                 avg_min_temp, count_min_temp, avg_humidity, count_humidity):
+
+        if dic["Max TemperatureC"] != "":
+            avg_max_temp += int(dic["Max TemperatureC"])  # adding up all the temperature occurrences
+            count_max_temp += 1
+
+        if dic["Min TemperatureC"] != "":
+            avg_min_temp += int(dic["Min TemperatureC"])
+            count_min_temp += 1
+
+        if dic[" Mean Humidity"] != "":  # lst[3] contains mean humidity results in a line
+            avg_humidity += int(dic[" Mean Humidity"])
+            count_humidity += 1
+
+        return avg_max_temp, count_max_temp, avg_min_temp, count_min_temp, avg_humidity, count_humidity
+
+    def average_per_month(self, year, month, avg_max_temp, count_max_temp,
+                          avg_min_temp, count_min_temp, avg_humidity, count_humidity):
+
+        file = self.files_path_prefix + year + "_" + month + ".txt"
+
+        if path.exists(file):
+            reader = CSVReader(file)
+
+            dictionary = reader.get_line()
+            while dictionary:
+                avg_max_temp, count_max_temp, avg_min_temp, \
+                    count_min_temp, avg_humidity, count_humidity \
+                    = self.calculate_average_helper(dictionary, avg_max_temp, count_max_temp,
+                                                    avg_min_temp, count_min_temp, avg_humidity, count_humidity)
+                dictionary = reader.get_line()
+
+            return True, avg_max_temp, count_max_temp, avg_min_temp, count_min_temp, avg_humidity, count_humidity
+        return False, avg_max_temp, count_max_temp, avg_min_temp, count_min_temp, avg_humidity, count_humidity
+
+    def calculate_average(self, date):
```

Comment:
According to the document we only need to calculate the average for a given month no need to calculate it for a year. Do the same for other reports as well.

Reply:
done